llvm/lib/IR/AutoUpgrade.cpp

```diff
@@ -5022,6 +5022,69 @@ bool llvm::UpgradeDebugInfo(Module &M) {
   return Modified;
 }
 
+bool static upgradeSingleNVVMAnnotation(GlobalValue *GV, StringRef K,
+                                        const Metadata *V) {
+  if (K == "kernel") {
+    assert(mdconst::extract<ConstantInt>(V)->getZExtValue() == 1);
```

Comment:
Assert should not be used here as it's not an LLVM error. It's an input error and should be diagnosed (or accepted/ignored). Considering that the value is a flag, and not very useful one at that (I don't think `kernel 0` is ever actually used, it would be OK to accept a non-zero value as "it's a kernel" and zero as "not a kernel".

Reply:
Sounds good, fixed